crypto/bio/b_addr.c

```diff
@@ -603,12 +603,16 @@ static int addrinfo_wrap(int family, int socktype,
 
 DEFINE_RUN_ONCE_STATIC(do_bio_lookup_init)
 {
-    if (!OPENSSL_init_crypto(0, NULL))
-        return 0;
     bio_lookup_lock = CRYPTO_THREAD_lock_new();
     return bio_lookup_lock != NULL;
 }
 
+static int bio_lookup_init(void)
+{
+    return OPENSSL_init_crypto(0, NULL)
```

Comment:
Because I found that pthread_once() uses some kind of locking mechanism, so if the same `pthread_once_t` is tripped again while its function is already running, I get a lock-up.  That may happen because of the `OPENSSL_init_crypto()` call...

Reply:
More, my question was why do we need the `OPENSSL_init_crypto()` call at all?

Reply:
In this spot, I actually do not know.  Lock creation doesn't need any further init stuff as far as I know.  So I can honestly not remember why it was put there in the first place.
(this leads me to think that there should really be a comment near each `OPENSSL_init_crypto` we do internally, with a reasoning why it's needed there)